File: app/src/types/sensibilidade.ts
export interface ReferenciaSensibilidade {
  cenario_base: string
  estrategicos_base: number | null
  quantidade_cenarios: number
}

export interface CenarioSensibilidadeResumo {
  cenario: string
  estrategicos_cenario: number | null
  estrategicos_em_comum_com_base: number | null
  retencao_dos_225_base_pct: number | null
  jaccard_pct: number | null
  spearman_relevancia_vs_base: number | null
  spearman_pressao_vs_base: number | null
}

export interface SensibilidadeResumoResponse {
  referencia: ReferenciaSensibilidade
  cenarios: CenarioSensibilidadeResumo[]
}

File: backend/app/schemas/sensibilidade.py
from pydantic import BaseModel


class ReferenciaSensibilidadeResponse(BaseModel):
    cenario_base: str
    estrategicos_base: int | None
    quantidade_cenarios: int


class CenarioSensibilidadeResumo(BaseModel):
    cenario: str
    estrategicos_cenario: int | None
    estrategicos_em_comum_com_base: int | None
    retencao_dos_225_base_pct: float | None
    jaccard_pct: float | None
    spearman_relevancia_vs_base: float | None
    spearman_pressao_vs_base: float | None


class SensibilidadeResumoResponse(BaseModel):
    referencia: ReferenciaSensibilidadeResponse
    cenarios: list[CenarioSensibilidadeResumo]
